app/actors/hydranorth/generic_file/actor.rb

```diff
@@ -18,6 +18,8 @@ def update_metadata(attributes, visibility)
         date_created = attributes[:date_created]
         generic_file.year_created  = date_created[/(\d\d\d\d)/,0] unless date_created.nil? || date_created.blank?
 
+        # TODO Do this much smarter! If you just hit an update from the UI,
+        # all this logic runs even though nothing changes! Should be checking for a diff of the ID's?
         ['hasCollectionId', 'belongsToCommunity'].each do |attr|
```

Comment:
I noticed this code block was executing on every update of a GenericFile from the UI. This is because when a user hits the update button, it will serialise the entire form and all its attributes and send it to our controller. We simply make the assumption here that if the attributes contains hasCollectionID or belongsToCommunity then they must have changed. But this simply isn't the true.

This does a lot of IO to the database (a few lookups and a few saves), which is expensive with Fedora. We should be doing a simple check to see if they actually changed before proceeding any further.